File: Data Structures & Algorithms/binary-tree-vertical-order-traversal/submission-1.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution {
    public List<List<int>> VerticalOrder(TreeNode root) {
        if (root == null) return new List<List<int>>();

        var cols = new SortedDictionary<int, List<int>>();
        var queue = new Queue<(TreeNode node, int pos)>();
        queue.Enqueue((root, 0));

        while (queue.Count > 0) {
            var (node, pos) = queue.Dequeue();

            if (!cols.ContainsKey(pos))
                cols[pos] = new List<int>();
            cols[pos].Add(node.val);

            if (node.left != null) queue.Enqueue((node.left, pos - 1));
            if (node.right != null) queue.Enqueue((node.right, pos + 1));
        }

        return cols.Values.ToList<List<int>>();
    }
}

File: Data Structures & Algorithms/valid-binary-search-tree/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    left_check = staticmethod(lambda val, limit: val < limit) 
    right_check = staticmethod(lambda val, limit: val > limit) 

    def isValidBST(self, root: Optional[TreeNode]) -> bool:
        if not root:
            return True
        
        return (self.isValid(root.left, root.val, self.left_check) and
                self.isValid(root.right, root.val, self.right_check))

    def isValid(self, root: Optional[TreeNode], limit: int, check) -> bool:
        if not root:
            return True
        if not check(root.val, limit):
            return False
        return (self.isValid(root.left, limit, check) and
                self.isValid(root.right, limit, check))
